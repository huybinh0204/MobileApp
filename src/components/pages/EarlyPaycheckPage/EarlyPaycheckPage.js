import { Dynatrace } from '@dynatrace/react-native-plugin';
import { useTheme } from '@emotion/react';
import Clipboard from '@react-native-community/clipboard';
import { useNavigation } from '@react-navigation/native';
import React, { useEffect, useRef, useState } from 'react';
import ENV from 'react-native-config';
import BottomSheet from 'react-native-raw-bottom-sheet';
import { useDispatch, useSelector } from 'react-redux';
import { AccountNumber, MainButton, Toast } from '_components/atoms';
import { SecondaryScreenLayout } from '_components/organisms';
import { ATOMIC_TYPES, ICONS, NAVIGATION, TOAST_TYPES } from '_constants';
import strings from '_localization';
import AtomicService from '_services/AtomicService';
import { AccountActions, AccountSelectors } from '_store/account';
import {
  FundDirectDepositStoreActions,
  FundDirectDepositStoreSelectors,
} from '_store/pageStore/fundDirectDepositStore';
import { normalize } from '_utilities/screen';
import {
  BottomSheetButton,
  BottomSheetDescription,
  BottomSheetTitle,
  Container,
  DirectDepositBottomSheetContainer,
  DisclaimerNote,
  DisclosureSheetContainer,
  DisclosureSheetContent,
  DisclosureSheetTitle,
  EarlyPaycheckContent,
  EarlyPaycheckTitle,
  FooterContent,
  Icon,
  LearnMoreLink,
  SubLink,
} from './EarlyPaycheckPage.styles';

const EarlyPaycheckPage = () => {
  const dispatch = useDispatch();
  const { navigate } = useNavigation();
  const { colors } = useTheme();

  const bottomSheetRef = useRef(null);
  const disclosureBottomSheet = useRef(null);
  const [toastText, setToastText] = useState(null);

  const externalId = useSelector(AccountSelectors.getExternalId);
  const directDepositFormError = useSelector(AccountSelectors.getDirectDepositFormError);
  const accountInfo = useSelector(FundDirectDepositStoreSelectors.getRenderData);
  const isLoading = useSelector(AccountSelectors.getIsLoadingAtomicFiToken);
  const token = useSelector(AccountSelectors.getAtomicFiToken);

  const { accountNumber, routingNumber } = accountInfo;

  useEffect(() => {
    if (externalId && (!accountNumber || !routingNumber)) {
      dispatch(FundDirectDepositStoreActions.fetchRenderData(externalId));
    }
  }, [dispatch, accountNumber, routingNumber, externalId]);

  useEffect(() => {
    if (externalId) {
      dispatch(AccountActions.fetchAtomicFiToken(externalId));
    }
  }, [externalId, dispatch]);

  const handleDirectDeposit = () => {
    bottomSheetRef.current?.open();
  };

  const handleGetForm = () => {
    bottomSheetRef.current?.close();
    navigate(NAVIGATION.shared.directDepositFormView);
  };

  const closeToast = () => {
    setToastText(null);
    dispatch(AccountActions.setDirectDepositFormError(false));
  };

  const handleContinue = () => {
    AtomicService.transact({
      token,
      brandColor: colors.alpha500,
      onInteraction(interaction) {
        if (interaction.name === ATOMIC_TYPES.COMPLETE_PAYCHECK) {
          navigate(NAVIGATION.shared.earlyPaycheckSuccess);
        } else if (interaction.name === ATOMIC_TYPES.AUTHENTICATION_ERROR) {
          Dynatrace.reportError(`AtomicFi Authentication Error: ${interaction.name}`, 0);
        }
      },
    });
  };

  const closeDisclosure = () => {
    disclosureBottomSheet.current?.close();
  };

  const openDisclosure = () => {
    disclosureBottomSheet.current?.open();
  };

  const copyToClipboard = ({ accountNumber: account, routingNumber: routing }) => {
    Clipboard.setString(account || routing);
    setToastText(
      account ? strings.accountInfo.accountNumberCopied : strings.accountInfo.routingNumberCopied
    );
  };

  return (
    <SecondaryScreenLayout>
      <Container>
        <Icon icon={ICONS.earlyPaycheck} width={normalize(128)} height={normalize(128)} />
        <EarlyPaycheckTitle>{strings.earlyPaycheck.title}</EarlyPaycheckTitle>
        <EarlyPaycheckContent>{strings.earlyPaycheck.description}</EarlyPaycheckContent>
        <FooterContent>
          <DisclaimerNote>
            {strings.formatString(strings.earlyPaycheck.disclaimer, {
              earlyDepositLink: (
                <LearnMoreLink onPress={openDisclosure}>
                  {strings.earlyPaycheck.earlyDepositText}
                </LearnMoreLink>
              ),
              atomicPrivacyLink: (
                <SubLink href={ENV.ATOMIC_FI_PRIVACY_POLICY}>
                  {strings.earlyPaycheck.atomicPrivacyText}
                </SubLink>
              ),
            })}
          </DisclaimerNote>
          <MainButton onPress={handleContinue} isLoading={isLoading}>
            {strings.earlyPaycheck.buttonText}
          </MainButton>
          <MainButton variant="link" onPress={handleDirectDeposit}>
            {strings.earlyPaycheck.setUpDirectDeposit}
          </MainButton>
        </FooterContent>
      </Container>
      <BottomSheet height={normalize(350)} ref={disclosureBottomSheet}>
        <DisclosureSheetContainer>
          <DisclosureSheetTitle>{strings.earlyPaycheck.disclosureTitle}</DisclosureSheetTitle>
          <DisclosureSheetContent>{strings.earlyPaycheck.disclosureBody}</DisclosureSheetContent>
          <MainButton onPress={closeDisclosure}>
            {strings.earlyPaycheck.disclosureButtonText}
          </MainButton>
        </DisclosureSheetContainer>
      </BottomSheet>
      <BottomSheet closeOnPressMask height={normalize(350)} ref={bottomSheetRef}>
        <DirectDepositBottomSheetContainer>
          <BottomSheetTitle>{strings.earlyPaycheck.bottomSheet.title}</BottomSheetTitle>
          <BottomSheetDescription>{strings.earlyPaycheck.bottomSheet.body}</BottomSheetDescription>
          <AccountNumber onPressAccount={copyToClipboard} onPressRouting={copyToClipboard} />
          <BottomSheetButton onPress={handleGetForm} testID="earlyPaycheckGetFormButton">
            {strings.earlyPaycheck.bottomSheet.buttonText}
          </BottomSheetButton>
        </DirectDepositBottomSheetContainer>
        <Toast
          onClose={closeToast}
          testID="earlyPaycheckPageToast"
          header={strings.accountInfo.copied}
          content={toastText}
          show={toastText !== null}
          type={TOAST_TYPES.INFO}
        />
      </BottomSheet>
      <Toast
        onClose={closeToast}
        testID="earlyPaycheckPageToastError"
        header={strings.overdraft.sell.directDepositFormError.title}
        content={strings.overdraft.sell.directDepositFormError.description}
        show={directDepositFormError}
        type={TOAST_TYPES.ERROR}
      />
    </SecondaryScreenLayout>
  );
};

export default EarlyPaycheckPage;
