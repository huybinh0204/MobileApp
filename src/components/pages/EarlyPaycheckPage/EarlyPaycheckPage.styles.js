import styled from '@emotion/native';
import { BaseText, IconSvg, Link, MainButton } from '_components/atoms';

export const Container = styled.View(({ theme }) => ({
  padding: theme.spacing.m,
  flex: 1,
}));

export const Icon = styled(IconSvg)(({ theme }) => ({
  alignSelf: 'center',
  marginTop: theme.spacing.m,
}));

export const EarlyPaycheckTitle = styled(BaseText)(({ theme }) => ({
  ...theme.typography.heading3,
  color: theme.colors.beta900,
  marginBottom: theme.spacing.xs,
  marginTop: theme.spacing.m,
  textAlign: 'center',
}));

export const EarlyPaycheckContent = styled(BaseText)(({ theme }) => ({
  ...theme.typography.subContentRegular,
  color: theme.colors.beta500,
  textAlign: 'center',
  paddingHorizontal: theme.spacing.s,
}));

export const FooterContent = styled.View({
  marginTop: 'auto',
});

export const DisclaimerNote = styled(BaseText)(({ theme }) => ({
  ...theme.typography.legal,
  color: theme.colors.beta500,
  marginBottom: theme.spacing.s,
  textAlign: 'center',
}));

export const LearnMoreLink = styled(BaseText)({
  textDecorationLine: 'underline',
});

export const DisclosureSheetContainer = styled.View(({ theme }) => ({
  padding: theme.spacing.m,
}));

export const DisclosureSheetTitle = styled(BaseText)(({ theme }) => ({
  ...theme.typography.heading6,
  color: theme.colors.beta900,
  marginBottom: theme.spacing.xs,
}));

export const DisclosureSheetContent = styled(BaseText)(({ theme }) => ({
  ...theme.typography.mainContentRegular,
  color: theme.colors.beta500,
  marginBottom: theme.spacing.m,
}));

export const SubLink = styled(Link)({
  textDecorationLine: 'underline',
});

export const BottomSheetTitle = styled(BaseText)(({ theme }) => ({
  ...theme.typography.heading5,
  color: theme.colors.beta900,
  marginBottom: theme.spacing.xs,
}));

export const BottomSheetDescription = styled(BaseText)(({ theme }) => ({
  ...theme.typography.mainContentRegular,
  color: theme.colors.beta400,
  marginBottom: theme.spacing.m,
}));

export const DirectDepositBottomSheetContainer = styled.View(({ theme }) => ({
  flex: 1,
  padding: theme.spacing.m,
}));

export const BottomSheetButton = styled(MainButton)(({ theme }) => ({
  marginVertical: theme.spacing.m,
}));
