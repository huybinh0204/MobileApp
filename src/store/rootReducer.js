import { combineReducers } from 'redux';
import actionWatchReducer from 'redux-action-watch/lib/reducer';
import { resettableReducer } from 'reduxsauce';
import { AccountReducer } from './account';
import { AuthenticationReducer } from './authentication';
import { CustomerReducer } from './customer';
import { FundDirectDepositStoreReducer } from './pageStore/fundDirectDepositStore';
import { RegisterReducer } from './register';
import { TransferReducer } from './transfer';

const resettable = resettableReducer('RESET');

export default combineReducers({
  account: resettable(AccountReducer),
  auth: resettable(AuthenticationReducer),
  customer: resettable(CustomerReducer),
  fundDirectDepositStoreReducer: resettable(FundDirectDepositStoreReducer),
  register: RegisterReducer,
  transfer: resettable(TransferReducer),
  watcher: actionWatchReducer, // Must be last
});
