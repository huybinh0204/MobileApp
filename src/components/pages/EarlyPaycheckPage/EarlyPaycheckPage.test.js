import { useNavigation } from '@react-navigation/native';
import { fireEvent, render } from '@testing-library/react-native';
import React from 'react';
import { useDispatch } from 'react-redux';
import { createStore } from 'redux';
import { NAVIGATION } from '_constants';
import strings from '_localization';
import rootReducer from '_store/rootReducer';
import ComponentWithProviders from '_test/mocks/ComponentWithProviders';
import { navigationMock } from '_test/mocks/navigation';
import EarlyPaycheckPage from './EarlyPaycheckPage';

jest.mock('@react-navigation/native', () => ({
  ...jest.requireActual('@react-navigation/native'),
  useNavigation: jest.fn(),
}));

jest.mock('react-redux', () => ({
  ...jest.requireActual('react-redux'),
  useDispatch: jest.fn(),
}));

describe('EarlyPaycheckPage', () => {
  const mockedDispatch = jest.fn();
  const mockedNavigation = navigationMock(jest.fn);

  beforeEach(() => {
    useDispatch.mockReturnValue(mockedDispatch);
    useNavigation.mockReturnValue(mockedNavigation);
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should match snapshot', () => {
    const { toJSON } = render(<ComponentWithProviders component={EarlyPaycheckPage} />);

    expect(toJSON()).toMatchSnapshot();
  });

  it('should navigate to the direct deposit form view', () => {
    const mockStore = createStore(rootReducer, {
      account: {
        directDepositFormError: false,
      },
    });

    const { getByTestId, getByText } = render(
      <ComponentWithProviders store={mockStore} component={EarlyPaycheckPage} />
    );

    const directDepositButton = getByText(strings.earlyPaycheck.setUpDirectDeposit);
    fireEvent.press(directDepositButton);

    const formButton = getByTestId('earlyPaycheckGetFormButton');
    fireEvent.press(formButton);

    expect(mockedNavigation.navigate).toHaveBeenCalledWith(NAVIGATION.shared.directDepositFormView);
  });

  it('should show toast error if fetch form fails', () => {
    const mockStore = createStore(rootReducer, {
      account: {
        directDepositFormError: true,
      },
    });

    const { getByTestId } = render(
      <ComponentWithProviders store={mockStore} component={EarlyPaycheckPage} />
    );

    const toast = getByTestId('earlyPaycheckPageToastError');

    expect(toast).not.toBeNull();
  });
});
