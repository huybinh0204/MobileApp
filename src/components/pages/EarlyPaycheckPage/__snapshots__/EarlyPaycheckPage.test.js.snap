// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EarlyPaycheckPage should match snapshot 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RNCSafeAreaProvider
    onInsetsChange={[Function]}
    style={
      Array [
        Object {
          "flex": 1,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "rgb(242, 242, 242)",
            "flex": 1,
          },
          undefined,
        ]
      }
    >
      <RNSScreenContainer
        onLayout={[Function]}
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <RNSScreen
          activityState={2}
          collapsable={false}
          forwardedRef={[Function]}
          pointerEvents="box-none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        >
          <View
            collapsable={false}
            style={
              Object {
                "opacity": 1,
              }
            }
          />
          <View
            accessibilityElementsHidden={false}
            closing={false}
            gestureVelocityImpact={0.3}
            importantForAccessibility="auto"
            onClose={[Function]}
            onGestureBegin={[Function]}
            onGestureCanceled={[Function]}
            onGestureEnd={[Function]}
            onOpen={[Function]}
            onTransition={[Function]}
            pointerEvents="box-none"
            style={
              Array [
                Object {
                  "display": "flex",
                  "overflow": undefined,
                },
                Object {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                },
              ]
            }
            transitionSpec={
              Object {
                "close": Object {
                  "animation": "spring",
                  "config": Object {
                    "damping": 500,
                    "mass": 3,
                    "overshootClamping": true,
                    "restDisplacementThreshold": 10,
                    "restSpeedThreshold": 10,
                    "stiffness": 1000,
                  },
                },
                "open": Object {
                  "animation": "spring",
                  "config": Object {
                    "damping": 500,
                    "mass": 3,
                    "overshootClamping": true,
                    "restDisplacementThreshold": 10,
                    "restSpeedThreshold": 10,
                    "stiffness": 1000,
                  },
                },
              }
            }
          >
            <View
              collapsable={false}
              pointerEvents="box-none"
              style={
                Object {
                  "flex": 1,
                  "marginTop": 0,
                }
              }
            >
              <View
                collapsable={false}
                forwardedRef={[Function]}
                needsOffscreenAlphaCompositing={false}
                onGestureHandlerEvent={[Function]}
                onGestureHandlerStateChange={[Function]}
                style={
                  Object {
                    "flex": 1,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                      Object {
                        "translateX": 0,
                      },
                    ],
                  }
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "flex": 1,
                        "overflow": "hidden",
                      },
                      Array [
                        Object {
                          "backgroundColor": "rgb(242, 242, 242)",
                        },
                        undefined,
                      ],
                    ]
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                        "flexDirection": "column-reverse",
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "flex": 1,
                        }
                      }
                    >
                      <RCTSafeAreaView
                        backgroundColor={null}
                        emulateUnlessSupported={true}
                        style={
                          Array [
                            Object {
                              "backgroundColor": "#FAFAFA",
                              "flex": 1,
                            },
                            undefined,
                          ]
                        }
                        testID="SecondaryScreenLayout"
                      >
                        <View
                          style={
                            Array [
                              Object {
                                "alignItems": "center",
                                "height": 108.5,
                                "justifyContent": "center",
                              },
                              undefined,
                            ]
                          }
                        >
                          <View
                            accessible={true}
                            collapsable={false}
                            focusable={true}
                            onClick={[Function]}
                            onResponderGrant={[Function]}
                            onResponderMove={[Function]}
                            onResponderRelease={[Function]}
                            onResponderTerminate={[Function]}
                            onResponderTerminationRequest={[Function]}
                            onStartShouldSetResponder={[Function]}
                            style={
                              Object {
                                "left": 36,
                                "opacity": 1,
                                "position": "absolute",
                                "zIndex": 10,
                              }
                            }
                            testID="BackButton"
                          >
                            <SvgMock
                              height={45.5}
                              width={45.5}
                            />
                          </View>
                        </View>
                        <View
                          onLayout={[Function]}
                          style={
                            Array [
                              Array [
                                Object {
                                  "flex": 1,
                                },
                                undefined,
                              ],
                              Object {
                                "paddingBottom": 0,
                              },
                            ]
                          }
                        >
                          <View
                            style={
                              Array [
                                Object {
                                  "flex": 1,
                                  "padding": 58,
                                },
                                undefined,
                              ]
                            }
                          >
                            <SvgMock
                              height={232}
                              style={
                                Array [
                                  Object {
                                    "alignSelf": "center",
                                    "marginTop": 58,
                                  },
                                  undefined,
                                ]
                              }
                              width={232}
                            />
                            <Text
                              allowFontScaling={false}
                              style={
                                Array [
                                  Object {
                                    "color": "#1F1D25",
                                    "fontFamily": "Poppins-SemiBold",
                                    "fontSize": 50.5,
                                    "lineHeight": 70.5,
                                    "marginBottom": 14.5,
                                    "marginTop": 58,
                                    "textAlign": "center",
                                  },
                                  undefined,
                                ]
                              }
                            >
                              Get your paycheck up to 2 days earlier*
                            </Text>
                            <Text
                              allowFontScaling={false}
                              style={
                                Array [
                                  Object {
                                    "color": "#5F6A77",
                                    "fontFamily": "Poppins",
                                    "fontSize": 25.5,
                                    "lineHeight": 35.5,
                                    "paddingHorizontal": 29,
                                    "textAlign": "center",
                                  },
                                  undefined,
                                ]
                              }
                            >
                              Connect your paycheck to your Kinly account automatically. Kinly uses Atomic to change your payroll provider direct deposit location.
                            </Text>
                            <View
                              style={
                                Array [
                                  Object {
                                    "marginTop": "auto",
                                  },
                                  undefined,
                                ]
                              }
                            >
                              <Text
                                allowFontScaling={false}
                                style={
                                  Array [
                                    Object {
                                      "color": "#5F6A77",
                                      "fontFamily": "Poppins",
                                      "fontSize": 20,
                                      "lineHeight": 28,
                                      "marginBottom": 29,
                                      "textAlign": "center",
                                    },
                                    undefined,
                                  ]
                                }
                              >
                                *Learn more about [object Object].
By clicking continue you agree to [object Object]
                              </Text>
                              <View
                                accessibilityState={
                                  Object {
                                    "disabled": false,
                                  }
                                }
                                accessible={true}
                                collapsable={false}
                                focusable={true}
                                onClick={[Function]}
                                onResponderGrant={[Function]}
                                onResponderMove={[Function]}
                                onResponderRelease={[Function]}
                                onResponderTerminate={[Function]}
                                onResponderTerminationRequest={[Function]}
                                onStartShouldSetResponder={[Function]}
                                style={
                                  Object {
                                    "alignSelf": "stretch",
                                    "backgroundColor": "#3D3397",
                                    "borderRadius": 10,
                                    "opacity": 1,
                                    "paddingHorizontal": 45.5,
                                    "paddingVertical": 31,
                                  }
                                }
                              >
                                <Text
                                  allowFontScaling={false}
                                  disabled={false}
                                  style={
                                    Array [
                                      Object {
                                        "color": "#FFFFFF",
                                        "fontFamily": "Poppins-SemiBold",
                                        "fontSize": 25.5,
                                        "lineHeight": 35.5,
                                        "textAlign": "center",
                                      },
                                      undefined,
                                    ]
                                  }
                                >
                                  Continue
                                </Text>
                              </View>
                              <View
                                accessibilityState={
                                  Object {
                                    "disabled": false,
                                  }
                                }
                                accessible={true}
                                collapsable={false}
                                focusable={true}
                                onClick={[Function]}
                                onResponderGrant={[Function]}
                                onResponderMove={[Function]}
                                onResponderRelease={[Function]}
                                onResponderTerminate={[Function]}
                                onResponderTerminationRequest={[Function]}
                                onStartShouldSetResponder={[Function]}
                                style={
                                  Object {
                                    "alignSelf": "stretch",
                                    "backgroundColor": "transparent",
                                    "borderRadius": 10,
                                    "opacity": 1,
                                    "paddingHorizontal": 45.5,
                                    "paddingVertical": 31,
                                  }
                                }
                              >
                                <Text
                                  allowFontScaling={false}
                                  disabled={false}
                                  style={
                                    Array [
                                      Object {
                                        "color": "#3D3397",
                                        "fontFamily": "Poppins-SemiBold",
                                        "fontSize": 25.5,
                                        "lineHeight": 35.5,
                                        "textAlign": "center",
                                      },
                                      undefined,
                                    ]
                                  }
                                >
                                  Other ways to set up direct deposit
                                </Text>
                              </View>
                            </View>
                          </View>
                          <Modal
                            animationType="none"
                            hardwareAccelerated={false}
                            onRequestClose={[Function]}
                            supportedOrientations={
                              Array [
                                "portrait",
                                "portrait-upside-down",
                                "landscape",
                                "landscape-left",
                                "landscape-right",
                              ]
                            }
                            transparent={true}
                            visible={false}
                          />
                          <Modal
                            animationType="none"
                            hardwareAccelerated={false}
                            onRequestClose={[Function]}
                            supportedOrientations={
                              Array [
                                "portrait",
                                "portrait-upside-down",
                                "landscape",
                                "landscape-left",
                                "landscape-right",
                              ]
                            }
                            transparent={true}
                            visible={false}
                          />
                          <View
                            onLayout={[Function]}
                            paddingHorizontal={null}
                            style={
                              Array [
                                Object {
                                  "alignItems": "center",
                                  "bottom": 0,
                                  "paddingHorizontal": 58,
                                  "position": "absolute",
                                  "width": "100%",
                                },
                                Object {
                                  "opacity": 0,
                                  "transform": Array [
                                    Object {
                                      "translateY": AnimatedValue {
                                        " __value": 0,
                                      },
                                    },
                                  ],
                                },
                              ]
                            }
                            testID="earlyPaycheckPageToastError"
                          >
                            <View
                              style={
                                Array [
                                  Object {
                                    "alignSelf": "stretch",
                                    "backgroundColor": "#FFFFFF",
                                    "borderColor": "#F2F3F4",
                                    "borderRadius": 10,
                                    "borderWidth": 0,
                                    "padding": 29,
                                    "shadowColor": "#362E60",
                                    "shadowOffset": Object {
                                      "height": 7,
                                      "width": 0,
                                    },
                                    "shadowOpacity": 0.1,
                                    "shadowRadius": 10,
                                  },
                                  Array [
                                    Object {
                                      "backgroundColor": "#D02B35",
                                      "borderColor": "#D02B35",
                                      "borderWidth": 1,
                                      "width": "100%",
                                    },
                                    undefined,
                                  ],
                                ]
                              }
                              type="error"
                            >
                              <View
                                style={
                                  Array [
                                    Object {
                                      "alignItems": "flex-start",
                                      "flexDirection": "row",
                                      "justifyContent": "space-between",
                                    },
                                    undefined,
                                  ]
                                }
                                testID="toastHeader"
                              >
                                <SvgMock
                                  height={36}
                                  width={36}
                                />
                                <Text
                                  allowFontScaling={false}
                                  numberOfLines={2}
                                  style={
                                    Array [
                                      Object {
                                        "color": "#FFFFFF",
                                        "flex": 1,
                                        "fontFamily": "Poppins-SemiBold",
                                        "fontSize": 21.5,
                                        "lineHeight": 30.5,
                                        "paddingHorizontal": 18,
                                        "paddingTop": 2,
                                      },
                                      undefined,
                                    ]
                                  }
                                  type="error"
                                >
                                  Issue retrieving direct deposit form
                                </Text>
                                <View
                                  accessibilityLabel="Close toast"
                                  accessibilityRole="button"
                                  accessible={true}
                                  collapsable={false}
                                  focusable={true}
                                  onClick={[Function]}
                                  onResponderGrant={[Function]}
                                  onResponderMove={[Function]}
                                  onResponderRelease={[Function]}
                                  onResponderTerminate={[Function]}
                                  onResponderTerminationRequest={[Function]}
                                  onStartShouldSetResponder={[Function]}
                                  style={
                                    Object {
                                      "opacity": 1,
                                    }
                                  }
                                  testID="toastCloseButton"
                                >
                                  <SvgMock
                                    height={20}
                                    style={
                                      Array [
                                        Object {
                                          "height": 65,
                                          "width": 65,
                                        },
                                        undefined,
                                      ]
                                    }
                                    type="error"
                                    width={20}
                                  />
                                </View>
                              </View>
                              <Text
                                allowFontScaling={false}
                                style={
                                  Array [
                                    Object {
                                      "color": "#FFFFFF",
                                      "fontFamily": "Poppins",
                                      "fontSize": 21.5,
                                      "lineHeight": 30.5,
                                      "paddingHorizontal": 54.5,
                                      "paddingTop": 3,
                                    },
                                    undefined,
                                  ]
                                }
                                type="error"
                              >
                                Please try again later
                              </Text>
                            </View>
                          </View>
                        </View>
                      </RCTSafeAreaView>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </RNSScreen>
      </RNSScreenContainer>
      <View
        collapsable={false}
        pointerEvents="box-none"
        style={
          Object {
            "height": 44,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "zIndex": 1,
          }
        }
      />
    </View>
  </RNCSafeAreaProvider>
</View>
`;
